website/docs/plugins/optuna_sweeper.md

```diff
@@ -0,0 +1,106 @@
+---
+id: optuna_sweeper
+title: Optuna Sweeper plugin
+sidebar_label: Optuna Sweeper plugin
+---
+
+[![Example application](https://img.shields.io/badge/-Example%20application-informational)](https://github.com/facebookresearch/hydra/tree/master/plugins/hydra_optuna_sweeper/example)
+[![Plugin source](https://img.shields.io/badge/-Plugin%20source-informational)](https://github.com/facebookresearch/hydra/tree/master/plugins/hydra_optuna_sweeper)
+
+
+This plugin enables Hydra applications to utilize [Optuna](https://optuna.org) for the optimization of the parameters of experiments.
+
+### Installation
+
+```commandline
+git clone https://github.com/facebookresearch/hydra.git
+cd hydra
+pip install plugins/hydra_optuna_sweeper
+```
+
+### Usage
+
+Please set `hydra/sweeper` to `optuna` in your config file.
+
+```yaml
+defaults:
+  - hydra/sweeper: optuna
+```
+
+Alternatively, add `hydra/sweeper=optuna` option to your command line.
+
+The default configuration is [here](https://github.com/facebookresearch/hydra/blob/master/plugins/hydra_optuna_sweeper/hydra_plugins/hydra_optuna_sweeper/config.py).
+
+### Example
+
```

Comment:
1. Could you add a section like nevergrad [here](https://hydra.cc/docs/next/plugins/nevergrad_sweeper#example-of-training-using-nevergrad-hyperparameter-search) on how to discover the params (python your_app hydra/sweeper=optuna --cfg hydra -p hydra.sweeper)
2. Could you add some doc on how to use the config file for params configuration?
3. Thx for adding the details on other override type! I'm wondering if we could format it a bit differently (by collapsing the example application command example and output), something like the following:



```
By default, `interval` is converted to [`UniformDistribution`](https://optuna.readthedocs.io/en/stable/reference/generated/optuna.distributions.UniformDistribution.html). You can use [`IntUniformDistribution`](https://optuna.readthedocs.io/en/stable/reference/generated/optuna.distributions.IntUniformDistribution.html) or [`LogUniformDistribution`](https://optuna.readthedocs.io/en/stable/reference/generated/optuna.distributions.LogUniformDistribution.html) by specifying the tags:


<details><summary>Example for interval override</summary>
...
</details>



`range` is converted to [`IntUniformDistribution`](https://optuna.readthedocs.io/en/stable/reference/generated/optuna.distributions.IntUniformDistribution.html). If you apply `shuffle` to `range`, [`CategoricalDistribution`](https://optuna.readthedocs.io/en/stable/reference/generated/optuna.distributions.CategoricalDistribution.html) is used instead.

<details><summary>Example for range override</summary>
...
</details>



```

here is an example on how it will be rendered https://hydra.cc/docs/next/plugins/ray_launcher#ray_aws-launcher

Reply:
Thank you for your suggestion. I applied your comments to the documentation in https://github.com/facebookresearch/hydra/pull/1132/commits/c342e8029be0eb4dd1d0dc18b919efdb9594fc4c. 

> 1. Could you add a section like nevergrad here on how to discover the params

I mentioned the parameter discovery at the head of the example section.

> 2. Could you add some doc on how to use the config file for params configuration?

I added the config file section after the override section.

> 3. I'm wondering if we could format it a bit differently (by collapsing the example application command example and output), something like the following:

That makes sense. Showing every output may be verbose. I applied `<detailes>` tag to the example code and outputs according to your suggestion.